### PR TITLE
Update list of deprecated methods

### PR DESCRIPTION
**Pull request by nickvergessen**
@DeepDiver1975 should get this into 8.2 so the occ check works as expected

@LukasReschke @MorrisJobke

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Looks good :+1:

**Comment by MorrisJobke** [comment]
And code works.

**Comment by rullzer** [comment]
Works as expected. :+1:

**Comment by MorrisJobke** [comment]
cc @DeepDiver1975

File: lib/private/util.php [hunk 1]
```diff
@@ -1300,12 +1300,11 @@ public static function generateRandomBytes($length = 30) {
 	/**
 	 * Get URL content
 	 * @param string $url Url to get content
-	 * @deprecated Use \OC::$server->getHTTPHelper()->getUrlContent($url);
 	 * @throws Exception If the URL does not start with http:// or https://
 	 * @return string of the response or false on error
 	 * This function get the content of a page via curl, if curl is enabled.
 	 * If not, file_get_contents is used.
-	 * @deprecated Use \OCP\Http\Client\IClientService
+	 * @deprecated Use \OC::$server->getHTTPHelper()->getUrlContent($url);
```
Comment (DeepDiver1975) [inline lib/private/util.php]: getHTTPHelper() is deprecated as well

Comment (MorrisJobke) [inline lib/private/util.php]: getHTTPClientService()->newClient()->get($url)

Comment (MorrisJobke) [inline lib/private/util.php]: @nickvergessen I updated it

**Comment by MorrisJobke** [comment]
Rebased to trigger CI

**Comment by nickvergessen** [comment]
Fixed the failing test

**Comment by MorrisJobke** [comment]
@nickvergessen PHP7:

```
1) Test\App\CodeChecker\CodeCheckerTest::testFindInvalidUsage with data set #3 ('OC_API', 1003, 'test-const.php')
Failed asserting that 0 matches expected 1.

/ssd/jenkins/workspace/server-master-linux-php7-ci/database/sqlite/label/SLAVE/tests/lib/app/codechecker/codecheckertest.php:30
```

**Comment by MorrisJobke** [comment]
> @DeepDiver1975 should get this into 8.2 so the occ check works as expected

cc @DeepDiver1975 @karlitschek

**Comment by karlitschek** [comment]
lets backport :+1:

**Pull request merged**